### PR TITLE
[IMP] mail: move default values outside of update

### PR DESCRIPTION
**Pull request by oomsveta**

### CODE REVIEW

File: addons/mail/static/src/new/core/thread_model.js [hunk 1]
```diff
@@ -30,69 +53,64 @@ export class Thread {
...
+            id: data.id,
+            type: data.type,
+            _state: state,
+        });
+        if (this.type === "channel") {
+            this._state.discuss.channels.threads.push(this.id);
+        } else if (this.type === "chat") {
+            this._state.discuss.chats.threads.push(this.id);
+        }
+        return this.update(data);
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js]: It doesn't look correct for update to return a value.

File: addons/mail/static/src/new/core/thread_model.js [hunk 2]
```diff
@@ -30,69 +53,64 @@ export class Thread {
...
         if (data.id in state.threads) {
             thread = state.threads[data.id];
+            thread.update(data);
         } else {
-            thread = new Thread();
-            thread._state = state;
+            thread = new Thread(state, data);
         }
-        thread.update(data);
         state.threads[thread.id] = thread;
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js]: only necessary to assign on the create case

File: addons/mail/static/src/new/core/thread_model.js [hunk 3]
```diff
@@ -30,69 +53,64 @@ export class Thread {
...
         // return reactive version
         return state.threads[thread.id];
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js:60]: accessing thread through reactive is only necessary in the create case too, insert "update" is already reading from state

**Comment by oomsveta** [comment]
@seb-odoo changes done

**Review by alexkuhn — APPROVED** [review]

File: addons/mail/static/src/new/core/thread_model.js [hunk 4]
```diff
@@ -27,72 +50,66 @@ export class Thread {
...
-        } else {
-            thread = new Thread();
-            thread._state = state;
+            const thread = state.threads[data.id];
+            thread.update(data);
+            return thread;
         }
-        thread.update(data);
-        state.threads[thread.id] = thread;
+        const thread = new Thread(state, data);
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/thread_model.js:58]: this no longer assign thread into state in the create case

Comment (seb-odoo) [reply addons/mail/static/src/new/core/thread_model.js:58]: actually it does but in constructor, this is a bit confusing to split the 2 state lines in different places

**Pull request merged**